src/number/format.js

```diff
@@ -68,6 +68,29 @@ return function( number, properties ) {
 		number *= 1000;
 	}
 
+	var compactPattern, compactDigits, compactProperties, divisor, pluralForm, rounded;
+	if ( compactMap ) {
+		var zeroes = Array(Math.floor(number).toString().length).join( "0" );
+		if ( zeroes.length >= 3 ) {
+			// use default plural form to perform initial decimal shift
+			compactPattern = compactMap[ "1" + zeroes + "-count-other" ];
+			compactDigits = compactPattern.split( "0" ).length - 1;
+			divisor = zeroes.length - ( compactDigits - 1 );
+			number = number / Math.pow( 10, divisor );
+
+			// calculate rounded value to determine plural form and extract correct pattern
+			rounded = parseInt(numberFormatIntegerFractionDigits( number, minimumIntegerDigits,
+				minimumFractionDigits, maximumFractionDigits, round, roundIncrement ));
+			pluralForm = pluralGenerator ? pluralGenerator(rounded) : "other";
+			compactPattern = compactMap[ "1" + zeroes + "-count-" + pluralForm ] || compactPattern;
+			compactProperties = compactPattern.match( numberPatternRe );
+
+			// update prefix/suffix with compact prefix/suffix
+			prefix += compactProperties[ 1 ];
+			suffix = compactProperties[ 11 ] + suffix;
+		}
+	}
```

Comment:
Like commented below, it would be ideal to use the biggest available instead of ignoring compact mode for big numbers.